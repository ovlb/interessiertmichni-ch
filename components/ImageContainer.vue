<template>
  <figure class="image-container">
    <img
      :src="imageSource"
      :alt="alt"
      class="image-container__image"
    >
    <image-link :img-id="imgId"/>
  </figure>
</template>

<script>
import ImageLink from './ImageLink.vue'

export default {
  props: {
    imgId: {
      type: Number,
      required: true
    },
    file: {
      type: String,
      required: true
    },
    alt: {
      type: String,
      required: true
    }
  },
  components: {
    ImageLink
  },
  computed: {
    imageSource () {
      return `/i/${this.file}`
    }
  }
}
</script>

<style>

.image-container {
  width: 100%;
}

.image-container__image {
  height: 240px;
  height: 75vmin;
  max-height: calc(.7 * 35rem);
  min-height: 240px;
  object-fit: contain;
  object-position: center;
  width: 100%;
}

.image-container__link::before {
  background: url(~assets/img/anchor.svg) center no-repeat;
  background-size: contain;
  content: "";
  display: block;
  flex: 0 0 .9rem;
  width: .9rem;
  height: .9rem;
  margin-right: .5rem;
}

</style>
