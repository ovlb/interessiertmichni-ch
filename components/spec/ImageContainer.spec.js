import { shallow } from '@vue/test-utils'
import ImageContainer from '../ImageContainer'
import testImageData from './util/testImageData'

describe('ImageContainer', () => {
  const wrapper = shallow(ImageContainer, {
    propsData: testImageData
  })

  test('receive props', () => {
    expect(wrapper.vm.file).toBe(testImageData.file)
  })

  test('construct the correct file src', () => {
    expect(wrapper.vm.imageSource).toBe(`/i/${testImageData.file}`)
  })

  test('match HTML structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
