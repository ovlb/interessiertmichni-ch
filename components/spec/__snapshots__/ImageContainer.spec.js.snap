// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageContainer match HTML structure 1`] = `
<figure
  class="image-container"
>
  <img
    alt="This is an alt text."
    class="image-container__image"
    src="/i/2341.gif"
  />
   
  <!---->
</figure>
`;
